geoh5py/ui_json/forms.py

```diff
@@ -0,0 +1,499 @@
+        value = IntegerParameter("value", value=value)
+        super().__init__(name, value=value, **kwargs)
+
+
+class FloatFormParameter(FormParameter):
+    """
+    Float parameter type.
+
+    :param min: Minimum value for ui element.
+    :param max: Maximum value for ui element.
+    :param precision: Number of decimal places to display in ui element.
+    :param line_edit: If False, the ui element incluces a spinbox.
+    """
+
+    identifier_members: list[str] = ["precision", "line_edit"]
+
+    def __init__(self, name, value=None, **kwargs):
+        self._min = FloatParameter("min")
+        self._max = FloatParameter("max")
+        self._precision = IntegerParameter("precision")
+        self._line_edit = BoolParameter("line_edit")
+        value = FloatParameter("value", value=value)
+        super().__init__(name, value=value, **kwargs)
+
+
+class ChoiceStringFormParameter(FormParameter):
+    """
+    Choice string parameter type.
+
+    :param choice_list: List of choices for ui dropdown.
+    """
+
+    identifier_members: list[str] = ["choice_list"]
+    validations = {"required_form_members": ["choice_list"]}
+
+    def __init__(self, name, value=None, **kwargs):
+        self._choice_list = StringListParameter("choice_list")
+        enforcers = None
+        if "choice_list" in kwargs:
+            enforcers = EnforcerPool(
+                "choice_list", [ValueEnforcer(kwargs["choice_list"])]
+            )
+        value = StringListParameter("value", value=value, enforcers=enforcers)
+        super().__init__(name, value=value, **kwargs)
+
+    def _add_value_enforcer(
+        self, choice_list: list, enforcers: EnforcerPool | None
+    ) -> EnforcerPool:
+        """
+        Updates enforcer pool to ensure parameter value in choice_list.
+
+        :param choice_list: list of ui element choices used for validation.
+        :param enforcers: Existing enforcer pool to update.
+        """
+        if enforcers is not None:
+            enforcers.enforcers.append(ValueEnforcer(choice_list))
+        else:
+            enforcers = EnforcerPool(self.name, [ValueEnforcer(choice_list)])
+
+        return enforcers
+
+
+class FileFormParameter(FormParameter):
+    """
+    File parameter type.
+
+    :param file_description: list of file descriptions for each file type.
+    :param file_type: list of file extensions to filter directory on.
+    :param file_multi: Allow multiple files to be selected from dropdown.
+    """
+
+    identifier_members: list[str] = ["file_description", "file_type", "file_multi"]
+    validations = {"required_form_members": ["file_description", "file_type"]}
+
+    def __init__(self, name, value=None, **kwargs):
+        self._file_description = StringListParameter("file_description")
+        self._file_type = StringListParameter("file_type")
+        self._file_multi = BoolParameter("file_multi")
+        value = StringParameter("value", value=value)

```

Comment:
Should probably turn this into a PathParameter eventually, similar behaviour as UUID where we can check if Path or str, and check if it exists.

Reply:
Exactly, it's on my list of Enforcers to add!